### ISSUE TITLE
request 1.35 LTS support

### ISSUE THREAD

**Issue by revansx** [krabina/Semantic-MediaWiki-Graph#2]
Tested this extension on MW 1.35 with no joy. There are no overt errors, but the extension neither shows up in Special:Version nor is there a new special page in the SMW section of Special:SpecialPages as indicated in the documentation. 

I tried setting requires": { "MediaWiki": ">= 1.38.0" } in extension.json to 1.35.0. It didn't generate any errors, but it didn't seem to have any effect either.

Any thoughts on how hard it would be to get it to run on 35.x (the current LTS version of MW?)

Thank you!
/Rich

**Comment by krabina** [comment]
maybe @slim16165 can help?

**Comment by slim16165** [comment]
Hi @revansx, have you added the call in the LocalSettings.php? It should be something like require_once "$IP/extensions/SemanticMediaWikiGraph/SemanticMediaWikiGraph.php";

If you already did, you could enable more debug tools:
i.e. 
$wgShowExceptionDetails = true;
$wgDevelopmentWarnings = true; 
$wgDebugToolbar = true;
error_reporting( -1 ); 
ini_set( 'display_errors', 1); 
error_reporting(E_ERROR);

**Comment by revansx** [comment]
As I am testing on MW 1.35 I used the wfLoadExtension( 'SemanticMediaWikiGraph' ); approach. 

Should I try the pre-MW1.24 method of installing the extension?

**Comment by krabina** [comment]
Here it seems to be installed and the special page is there, but not working:
https://www.tematichedigenere.com/wiki/Speciale:SemanticMediaWikiGraph

**Comment by slim16165** [comment]
> As I am testing on MW 1.35 I used the wfLoadExtension( 'SemanticMediaWikiGraph' ); approach.
> 
> Should I try the pre-MW1.24 method of installing the extension?

No, no. It should be equivalent. Can you try to see if enabling a more verbose log / error handling gives some hint?

> Here it seems to be installed and the special page is there, but not working: https://www.tematichedigenere.com/wiki/Speciale:SemanticMediaWikiGraph

I'm playing with a new release there. I'm trying to port everything to TypeScript

**Comment by revansx** [comment]
Special:SemanticMediaWikiGraph?requestDebug shows the following errors at the top of the page:
```
Notice
: Undefined index: name in
/opt/htdocs/mediawiki/includes/registration/ExtensionRegistry.php
on line
382

Notice
: Undefined index: name in
/opt/htdocs/mediawiki/includes/registration/ExtensionProcessor.php
on line
474

Notice
: Undefined index: name in
/opt/htdocs/mediawiki/includes/registration/ExtensionProcessor.php
on line
649

```
These errors are only present when the extension is installed

**Comment by slim16165** [comment]
> Special:SemanticMediaWikiGraph?requestDebug shows the following errors at the top of the page:
> These errors are only present when the extension is installed

Actually I've tried to install a fresh MW 1.35 on my server and then the extension and added it to the LocalSettings.php. I checked the path is resolved but anyway the extension doesn't work and doesn't give any warning. 

I suspect this could be a bug from MW. I don't expect this to be a desired behaviour... unless it's related to the version of PHP on that server; it's PHP 8, maybe this creates an extra issue? 

I can share the SSH access to the server if someone wants to do a test.

@krabina @revansx
